backend/layers/persistence/persistence.py

```diff
@@ -92,6 +92,12 @@ def _manage_session(self, **kwargs):
             if session is not None:
                 session.close()
 
+    @contextmanager
+    def _get_serializable_session(self, **kwargs):
+        with self._manage_session(**kwargs) as session:
+            session.connection(execution_options={"isolation_level": "SERIALIZABLE"})
```

Comment:
What does "SERIALIZABLE" do

Reply:
Effectively, it prevents the race condition of two parallel db sessions clobbering one another's writes due each reading the same data, computing a result with that data,  and then writing back the result, where the losing process hasn't incorporated the winning process's update. The winning process's write gets clobbered.

https://www.postgresql.org/docs/current/transaction-iso.html#XACT-SERIALIZABLE

Postgres is actually quite smart about this mode and doesn't actually run the 2 txns in a serial manner--instead they run in parallel, but psql figures out if a "serializable anomaly" has occurred and errors out the offending process.

Reply:
>Serializable isolation typically relies upon locking of rows or ranges of rows in order to achieve this effect and can increase the chance of deadlocks and degrade performance

"increase the chance of deadlocks" -- I don't see how/why this would be the case... 🤷‍♂️